mantidproject/mantid
# Fit Script Generator - Output Base Name and Plot Output options

**Description of work.**
This PR adds two options to the Fit Script Generator to be used when generating a python script:

- `Output Base Name` allows you to specify a different base name for your output fit workspaces
- `Plot Output` allows you to turn off/on the plotting of the output fit workspaces.

The PR also renames the `BasicFitOptionsBrowser` to be the `FitScriptOptionsBrowser` because it now contains an option specific to the fit script that you generate.

**To test:**
1. Comment out this line
https://github.com/mantidproject/mantid/blob/5077722a251590b9196563d8d23245ce15e04699/scripts/Muon/GUI/Common/fitting_widgets/basic_fitting/fit_controls_view.py#L28
1. Open Muon Analysis interface
2. Load MUSR 62260
3. Go to Fitting tab and click `Fit Generator`
4. When the `Fit Script Generator` opens, there should be 4 domains loaded into the interface
5. Add a `GausOsc` function to all domains
6. Change the `Output Base Name` (bottom right)
6. Click `Generate Script to Clipboard`.
7. Paste the script in a script editor window and run it. Make sure the script runs successfully, and the output workspaces have the expected base name.
8. Untick the `Plot Output` option and then re-generate the script. Run the script and make sure nothing is plotted.

9. Repeat this process but change the Fitting Mode to `Simultaneous` (bottom right)

Fixes #31955

*This does not require release notes* because **this interface is not yet visible to the user.**

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
